tsp-cli-client

```diff
@@ -310,17 +320,69 @@ if __name__ == "__main__":
                 print("Trace UUID is missing")
                 sys.exit(1)
 
+        if (options.get_xy_tree):
+            if options.uuid is not None:
+
+                output_descriptor = get_descriptor(
+                    options.uuid, options.get_xy_tree)
+                if (output_descriptor is None):
+                    sys.exit(1)
+
+                response = tsp_client.fetch_xy_tree(
+                    options.uuid, options.get_xy_tree)
+                if response.status_code == 200:
+                    tree = response.model.model
+                    if tree is None:
+                        print("Tree had no model; retry?")
+                        sys.exit(1)
+
+                    if (output_descriptor.type == "DATA_TREE"):
```

Comment:
> For xy charts to get the type is "XY_TIME_TREE", to indicate that it provides a tree and xy time series. The "DATA_TREE" type is meant for trees only with no XY time series or time graphs, for example used for statistics data.
> 
> When I tested it with the CpuUsageDataProvider, there is nothing printed. So, I don't see any IDs. To be used with the get-xy call.
> 
> The 3 types "DATA_TREE", "XY_TIME_TREE" and "TIMEGRAPH" provide a tree and the data structure returned is same. If you check for XY_TIME_TREE then you will get a formatted print-out using pandas. But that one doesn't print the internal IDs.
> 
> The option --get-tree was used before to also to print-out the DATA_TREE to test the function duration statistics data provider which is a DATA_TREE. In my initial implementation I didn't distinguish between TIMETRAPH or DATA_TREE. Sorry, for the confusion.

Thanks for the explanations. I just pushed a patch update that replaces DATA_TREE with TREE_TIME_XY (for the XY case). As for the TIME_GRAPH case, not sure how it should fit in this code for now. Maybe related to your next questions (right below). As for the IDs, I'm able to get some `Series id` printed out through the --get-xy command.

> I wonder if we should have on method for all 3 cases since the output data structure is the same? What do you think?

I'd need you to please back-log something that details your proposal, as I barely understand it with my current awareness of these aspects. I'm planning to resume working on test coverage for this library, so maybe that will lead me towards this goal.

Reply:
I meant the entry ID which are passed to the get-xy call as a list of items (`--items`). Right now due to some inconsistencies in different data provider implementations in the TC server and non-compliance to the TSP, it's not straight forward to give a recommendation to update the print method. I opened separate issues to track updates to fix that (#34, #33).